account/lnurl.go

```diff
@@ -35,11 +37,52 @@ func (a *Service) HandleLNURL(encodedLnurl string) (*data.LNUrlResponse, error)
 				},
 			},
 		}, nil
+	case lnurl.LNURLChannelResponse:
+		lsp := &lnurlLSP{&params}
+		err := a.openChannel(lsp, true)
```

Comment:
This function is called from "FetchLnurl" which implies read-only execution flow.
I find it a bit problematic that the client calls "FetchLnurl" and channel is opened.
I think it is better to create a function for each lnurl operation (like withdraw), this way the client knows exactly what he is asking for.
_OpenLnurlChannel_ was good in that sense.

Reply:
You mean creating another function on **breezmobile** that calls the same Go function?

What about renaming it to `HandleLnurl` or something like that that implies that _some lnurl-related action_ will happen?

Reply:
> What about renaming it to `HandleLnurl` or something like that that implies that _some lnurl-related action_ will happen?

My concern is that the url itslef doesn't actually tells the mobile user what is the _lnurl-related_ action this is why I think it should be more explicit and not one "HandleLnurl" for all.

Reply:
Oh, right, now I got it. You're thinking about the possibility of someone scanning an lnurl-channel from the lnurl-withdraw screen and ending up getting a channel instead of a withdraw.

Reply:
Yes exactly.
I think the way lnurl is designed is to always have two steps:
1. Fetch - to get the operation and metadata and get the user approval.
2. Finish - to execute the operation.

So in our case the Fetch (`HandleLNURL`) may behave the same as withdraw and return an LNUrlResponse_OpenChannel and in addition we can implement: `FinishLNURLOpenChannel` to execute the operation.
Following this pattern I believe will allow us implement all lnurl external links handling (including open channel) at the same point in breezmobile code and in a generic way.